### ISSUE TITLE
(AA) Headings and labels could be clearer

### ISSUE THREAD

**Issue by otis-bath** [open-sdg/open-sdg#1758]
**Medium level (AA)**

1. Headings and labels did not sufficiently describe the topic or purpose of the content. Screen reader users did not find the label ‘toggle navigation’ to sufficiently describe the purpose of the menu button. Additionally, as the intuitive associated name of the icon does not match the accessible name of the control, voice activation users may not be able to reference the control by the expected name, i.e., by speaking the command ‘click menu’.
![image](https://user-images.githubusercontent.com/90769108/182151996-8970e525-2515-4071-9d7d-34d40fa4a720.png)
Even a change like 'toggle navigation menu' would make this clearer in this instance.
![image](https://user-images.githubusercontent.com/90769108/182152105-9a57e0f5-aa93-488f-84d9-56c06b6d5389.png)

2. Another example is here with the 'edit' button. The users found it unclear what this tab would involve, whether this was editing data, metadata or the map. 
![image](https://user-images.githubusercontent.com/90769108/182158885-14657b19-7c02-4522-9ee3-cc2e5641eaec.png)
![image](https://user-images.githubusercontent.com/90769108/182158941-78ff4dc8-1f38-4af8-bb90-b3662b7a3c05.png)
'Edit data' was proposed as an alternative, we simply need more detail than just 'Edit'

3. Likewise, screen reader users did not find the label ‘Show search’ to sufficiently describe the purpose of the map search button. Screen reader users navigating ‘out of context’, filtering the page by element type, will have to switch navigational methods, and search through the surrounding page content in order to identify that this search button is related to the map and not the site search edit field.
![image](https://user-images.githubusercontent.com/90769108/182159242-bdfcd606-c670-4d0d-8971-42c1006f5959.png)
The screenreader user describes the issue well: “When attempting to locate the map search, I located information that was announced to me as ‘show search’, and ‘search edit’. This does not clear convey that the first search item relates to a site search whereas the second search item is specific to the map. I would expect to hear JAWS announce, ‘site search’ and ‘map search’ to indicate their individual functions more clearly.”
![image](https://user-images.githubusercontent.com/90769108/182159479-820e4414-4544-4aad-a91e-55b715597f2d.png)

We need the image labels here to be something that could intuitively match the icon.

**Comment by brockfanning** [comment]
@otis-bath We have some decisions to make with this one. Some possible ideas for each of the items mentioned:
1. "Toggle navigation menu"
2. "Edit this indicator"
3. "Show site-wide search"

**Comment by otis-bath** [comment]
Thanks @brockfanning - one for the tech call next week I suggest

**Comment by otis-bath** [comment]
Toggle navigation menu
Edit indicator 
Site search

**Comment by otis-bath** [comment]
Search -> Site search
---
Show search -> Show map search
Search -> Map search
Hide search -> Hide map search

**Comment by otis-bath** [comment]
Sarah to go through and check other buttons and capture those which we are using

**Comment by otis-bath** [comment]
@brockfanning 

This is the message we got back from Sarah: 

The title attribute is really only to provide users with additional information in a tooltip. These aren’t accessible to all users (such as keyboard only users, voice activation users and low vision users), so should really only be relied upon in instances where the information they are offering is supplementary and conveyed by other means. 

The content of the title attribute doesn’t have to match the accessible name/label of the control. However, where an aria-label is used, I assume that this is because you have used an icon as a visible label for the control? In this case it is nice to have the title offer a written version of the accessible name of the control (as specified by the aria-label). This can certainly assist users who may struggle to identify the meaning of icons. However, this is not necessary for conformance, and all information should be sufficiently clear without the title attribute being used at all.

**Comment by phillipgardiner** [comment]
@brock @LucyGwilliamAdmin @otis-bath as discussed passing on the spreadsheet that captures all the site changes required to aria labels and the alt text. Hopefully the spreadsheet is self explanatory but essentially on the left hand side is a list of all the components of the website and on the right are the required aria labels and alt text for those components. Any questions please give one of the team a shout.  (Sarah is back in the office on Thursday 15th).

[Hover_labels_library.xlsx](https://github.com/open-sdg/open-sdg/files/9558186/Hover_labels_library.xlsx)

**Comment by brockfanning** [comment]
Some comments/decisions/questions from last call.

[Hover_labels_library.xlsx](https://github.com/open-sdg/open-sdg/files/9656073/Hover_labels_library.xlsx)

**Comment by sarah-beasley** [comment]
@brockfanning DAC have confirmed we have done all we can to improve the maps. This is the [link](http://uk-sdg-feature-branches.s3-website.eu-west-2.amazonaws.com/feature-site-map-background-accessible2/3-3-3/) to the feature branch where the proposed map background changes have been made  - darkening the text, changing the water colour and changing the colour scale. Is this down to us on our end to implement now?
